core/dbt/materializations/incremental/microbatch.py

```diff
@@ -0,0 +1,130 @@
+from datetime import datetime, timedelta
+from typing import List, Optional, Tuple
+
+import pytz
+
+from dbt.artifacts.resources.types import BatchSize
+from dbt.contracts.graph.nodes import ModelNode, NodeConfig
+from dbt.exceptions import DbtInternalError, DbtRuntimeError
+
+
+class MicrobatchBuilder:
```

Comment:
I think we should add some docstrings to the class and functions in this file. It will set a good precedence, and also allows us to document some edge cases / assumptions that the functions make.